### PR TITLE
    Make template compatible with Proxmox 8.2.4, fix issues across template

### PR DESCRIPTION
**Pull request by oreze**
List of changes:

- fix, chore: upgrade providers, add proxmox >8.1 compatibility, fix breaking changes, adjust disk sizes and names of the templates
- fix: add missing cloudinit drive, enable automatic_reboot on modify operation
- fix: update docker script to wait if apt-get is locked
- fix: add missing quotes around ssh_private_key in kubespray setup
- chore: bump kubespray_image, kube_version, argocd_version versions, add explicit sesnitive properties to ssh keys variables

Basically, I've just upgraded providers, fixed some compatibility and other issues. It worked for me with Proxmox 8.2.4 with ubuntu 24.04 template.

I had to fix docker install script, it failed a few times because of apt-get lock, change in ```setup_kubespray.sh``` script should be reverted after merging this PR: [https://github.com/docker/docker-install/pull/431](https://github.com/docker/docker-install/pull/431)

Supersededs #90

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,8 +5,13 @@
 *.tfplan
 cloud.tf
 tmp
+*.tfvars
+
+# Exclude example.tfvars from being ignored
+!example.tfvars
```
Comment (oreze) [inline .gitignore:11]: wanted to exclude all .tfvars files from being checked, just in case

File: modules/proxmox_ubuntu_vm/main.tf [hunk 2]
```diff
@@ -27,27 +27,37 @@ resource "proxmox_vm_qemu" "ubuntu_vm" {
   scsihw           = "virtio-scsi-single"
   hotplug          = "network,disk,usb,memory,cpu"
   numa             = true
-  automatic_reboot = false
+  automatic_reboot = true
```
Comment (oreze) [inline modules/proxmox_ubuntu_vm/main.tf:30]: there were a problems with modify operations without this property set to true

File: modules/proxmox_ubuntu_vm/main.tf [hunk 3]
```diff
@@ -27,27 +27,37 @@ resource "proxmox_vm_qemu" "ubuntu_vm" {
...
-    slot     = 0
-    type     = "virtio"
-    storage  = var.vm_os_disk_storage
-    size     = "${var.vm_os_disk_size_gb}G"
-    iothread = 1
-  }
-
-  dynamic "disk" {
-    for_each = var.add_worker_node_data_disk ? [var.worker_node_data_disk_size] : []
+  disks {
```
Comment (oreze) [inline modules/proxmox_ubuntu_vm/main.tf:34]: new disks structure in updated provider

File: modules/proxmox_ubuntu_vm/main.tf [hunk 4]
```diff
@@ -27,27 +27,37 @@ resource "proxmox_vm_qemu" "ubuntu_vm" {
...
+        content {
+          disk {
+            size     = "${var.worker_node_data_disk_size}G"
+            storage  = var.worker_node_data_disk_storage
+            iothread = true
+          }
+        }
+      }
+    }
+    ide {
```
Comment (oreze) [inline modules/proxmox_ubuntu_vm/main.tf:55]: previous config missed cloudinit drive, therefore VMs were not initialized properly

File: scripts/setup_kubespray.sh [hunk 5]
```diff
@@ -42,6 +42,16 @@ if ! command -v docker &> /dev/null; then
         exit 1
     fi
 
+# Check and add DPkg::Lock::Timeout=600 to apt-get install if not already present
```
Comment (oreze) [inline scripts/setup_kubespray.sh:45]: Added simple script adding lock timeout, otherwise script would fail sometimes

File: vm-k8s-nodes.tf [hunk 6]
```diff
@@ -16,7 +16,7 @@ module "k8s_control_plane_nodes" {
   vm_net_subnet_cidr           = var.internal_net_subnet_cidr
   vm_host_number               = 10
   vm_user                      = var.vm_user
-  vm_tags                      = "${var.env_name};terraform;k8s-control-plane"
+  vm_tags                      = "${var.env_name};terraform;k8s_control_plane"
```
Comment (oreze) [inline vm-k8s-nodes.tf:19]: new naming convention, - is forbidden

File: vm-kubespray-host.tf [hunk 7]
```diff
@@ -80,7 +80,7 @@ resource "null_resource" "setup_kubespray" {
   provisioner "remote-exec" {
     inline = [
       local.setup_kubespray_script_content,
-      "echo ${var.ssh_private_key} | base64 -d > ${local.kubespray_data_dir}/id_rsa",
+      "echo \"${var.ssh_private_key}\" | base64 -d > ${local.kubespray_data_dir}/id_rsa",
```
Comment (oreze) [inline vm-kubespray-host.tf:83]: added missing "key" quotes, fails without it

**Review by khanh-ph — APPROVED** [review]
